### PR TITLE
chore: auto-detect React version in .eslintrc.yml

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Changes proposed in this pull request

Adds `settings.react.version = "detect"` to `.eslintrc.yml` so the following warning will no longer be logged from `yarn lint`:

```plaintext
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .
```

## Context

Per https://github.com/yannickcr/eslint-plugin-react#configuration, eslint-plugin-react doesn't automatically auto-detect the React version used. 

## Checklist

Please pardon the lack of related issue, I figured this small of a chore didn't warrant one. But I'm happy to file one if you'd prefer! 😄 

- [ ] Related issues linked using `fixes #number`
- ~[ ] Tests added/updated~
- ~[ ] Documentation added~
- [x] Make sure that all checks pass

### CODE REVIEW

File: jest.config.base.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
 module.exports = {
   globals: {
     'ts-jest': {
+      isolatedModules: true,
```
Comment (wilsonianb) [inline jest.config.base.js:6]: Is this related to the react version detection?

Comment (JoshuaKGoldberg) [reply jest.config.base.js:6]: Oh, shoot, sorry - didn't realize I'd left this in 😅. Removing.

**Review by wilsonianb — APPROVED** [review]
